src/Doctrine/DocumentRegenerator.php

```diff
@@ -0,0 +1,236 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Doctrine;
+
+use Doctrine\Common\Persistence\Mapping\MappingException as LegacyCommonMappingException;
+use Doctrine\ODM\MongoDB\Mapping\ClassMetadata;
+use Doctrine\ODM\MongoDB\Mapping\MappingException;
+use Doctrine\Persistence\Mapping\MappingException as PersistenceMappingException;
+use Symfony\Bundle\MakerBundle\Exception\RuntimeCommandException;
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\Util\ClassSourceManipulator;
+
+/**
+ * @internal
+ */
+final class DocumentRegenerator
+{
+    public function __construct(
+        private DoctrineODMHelper $doctrineHelper,
+        private FileManager $fileManager,
+        private Generator $generator,
+        private DocumentClassGenerator $documentClassGenerator,
+        private bool $overwrite,
+    ) {
+    }
+
+    public function regenerateDocuments(string $classOrNamespace): void
+    {
+        try {
+            $metadata = $this->doctrineHelper->getMetadata($classOrNamespace);
+        } catch (MappingException|LegacyCommonMappingException|PersistenceMappingException) {
+            $metadata = $this->doctrineHelper->getMetadata($classOrNamespace, true);
+        }
+
+        if ($metadata instanceof ClassMetadata) {
+            $metadata = [$metadata];
+        } elseif (class_exists($classOrNamespace)) {
+            throw new RuntimeCommandException(sprintf('Could not find Doctrine metadata for "%s". Is it mapped as a document?', $classOrNamespace));
+        } elseif (empty($metadata)) {
+            throw new RuntimeCommandException(sprintf('No documents were found in the "%s" namespace.', $classOrNamespace));
+        }
+
+        /** @var ClassSourceManipulator[] $operations */
+        $operations = [];
+        foreach ($metadata as $classMetadata) {
+            if (!class_exists($classMetadata->name)) {
+                // the class needs to be generated for the first time!
+                $classPath = $this->generateClass($classMetadata);
+            } else {
+                $classPath = $this->getPathOfClass($classMetadata->name);
+            }
+
+            $mappedFields = $this->getMappedFieldsInDocument($classMetadata);
+
+            if ($classMetadata->customRepositoryClassName) {
+                $this->generateRepository($classMetadata);
+            }
+
+            $manipulator = $this->createClassManipulator($classPath);
+            $operations[$classPath] = $manipulator;
+
+            $embeddedClasses = [];
+
+            foreach ($classMetadata->getEmbeddedFieldsMappings() as $fieldName => $mapping) {
+                if (str_contains($fieldName, '.')) {
+                    continue;
+                }
```

Comment:
While not recommended, field names in MongoDB can contain dots. Why is this excluded here?